src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -826,30 +811,21 @@ void SmartAI::SetCombatMove(bool on)
 {
     if (mCanCombatMove == on)
         return;
+
     mCanCombatMove = on;
+
     if (!IsAIControlled())
         return;
-    if (!HasEscortState(SMART_ESCORT_ESCORTING))
+
+    if (me->IsEngaged())
```

Comment:
`IsEngaged` guarantees combat. In addition, `IsEngaged` guarantees that the threat list is not empty for creatures that can have a threat list.